schematic/store/synapse.py

```diff
@@ -2108,16 +2106,35 @@ def _read_manifest(self, metadataManifestPath: str) -> pd.DataFrame:
 
     def _add_id_columns_to_manifest(
         self, manifest: pd.DataFrame, dmge: DataModelGraphExplorer
-    ):
-        """Helper function to add id and entityId columns to the manifest if they do not already exist, Fill id values per row.
-        Args:
-            Manifest loaded as a pd.Dataframe
-        Returns (pd.DataFrame):
-            Manifest df with new Id and EntityId columns (and UUID values) if they were not already present.
+    ) -> pd.DataFrame:
         """
+        Ensures that the manifest DataFrame has standardized 'Id' and 'entityId' columns.
+
+        If any case variation of the 'id' column is present (e.g., 'id', 'ID', 'iD'), it is renamed to 'Id'.
+        If any case variation of the 'entityid' column is present, it is renamed to 'entityId'.
+        If 'Id' is still missing, it will be created as an empty column or derived from a 'Uuid' column,
+        depending on whether 'Uuid' is defined in the schema.
+        Missing values in the 'Id' column are filled with generated UUIDs.
+        If 'entityId' is still missing, it will be created and filled with empty strings.
+        If it is already present, any missing values will be replaced with empty strings.
 
-        # Add Id for table updates and fill.
-        if not col_in_dataframe("Id", manifest):
+        Args:
+            manifest (pd.DataFrame): The metadata manifest to be updated.
+            dmge (DataModelGraphExplorer): data moodel graph explorer object
+
+        Returns:
+            pd.DataFrame: The updated manifest with a standardized 'Id' column and an 'entityId' column.
+        """
+        # Normalize any variation of 'id' to 'Id', "entityid" to "entityId"
+        for col in manifest.columns:
+            if col.lower() == "id":
+                manifest = manifest.rename(columns={col: "Id"})
+            if col.lower() == "entityid":
+                manifest = manifest.rename(columns={col: "entityId"})
+
```

Comment:
Since we might use the `uuid` column down below, should we also rename it here like you do with `id` and `entityid`

Reply:
addressed it per @SageGJ 's comment [here](https://github.com/Sage-Bionetworks/schematic/pull/1592#discussion_r2038267850)